### PR TITLE
[admission-policy-engine] add custom ca to verifyImageSignatures

### PR DESCRIPTION
**Pull request by miklezzzz**
## Description

Provides the patch that allows adding custom CAs to the list of trusted CAs of the ratify http client.

## Why do we need it, and what problem does it solve?

It enables a user to verify image signatures from registries secured with custom SSL certificates.

## What is the expected result?
It's possible to verify image signatures from private registries, protected by custom SSL certificates. The list of custom CAs can be set in a security policy.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [x] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: admission-policy-engine
type: feature
summary: Add custom CA for private registries when verifying image signatures.
impact_level: default
```

### CODE REVIEW

**Comment by miklezzzz** [comment]
before
```
"msg":"REPOSITORY_OPERATION_FAILURE: failed to resolve the subject descriptor: Unable to resolve the reference: nexus.dev.m-scherba.hf.flant.com:8089/security/myhello:latest: Head \"https://nexus.dev.m-scherba.hf.flant.com:8089/v2/security/myhello/manifests/latest\": tls: failed to verify certificate: x509: certificate signed by unknown authority (possibly because of \"crypto/rsa: verification error\" while trying to verify candidate authority certificate \"Internet Widgits Pty Ltd\")"
```

after
```
"msg":"verifying subject nexus.dev.m-scherba.hf.flant.com:8089/security/myhello:latest","namespace":"","time":"2024-12-03T21:37:24.873144153Z","trace-id":"f54af1b0-8b3f-41e6-9c0a-25928d9f9655"}
{"component-type":"executor","go.version":"go1.22.6","level":"info","msg":"Resolve of the image completed successfully the digest is sha256:a5e0f513db96448197a2a095d73ea7f0779c6730c762fc345c052079aba0fe51","namespace":"","time":"2024-12-03T21:37:24.886680125Z","trace-id":"f54af1b0-8b3f-41e6-9c0a-25928d9f9655"}
{"component-type":"server","go.version":"go1.22.6","level":"info","msg":"verification response for subject nexus.dev.m-scherba.hf.flant.com:8089/security/myhello:latest: \n{\n  \"version\": \"0.2.0\",\n  \"isSuccess\": true,\n
```

File: modules/015-admission-policy-engine/crds/security-policy.yaml [hunk 1]
```diff
@@ -458,6 +458,12 @@ spec:
                             description: The list of Cosign compliant public keys.
                             items:
                               type: string
+                          registryCAs:
```
Comment (yalosev) [inline modules/015-admission-policy-engine/crds/security-policy.yaml:461]: I don't think we need an array here.
Everywhere we have a `ca` field with a single value
```
ca:
  type: string
```
I think we have to follow this pattern here  to have a common flow

Comment (miklezzzz) [reply modules/015-admission-policy-engine/crds/security-policy.yaml]: disagree, but whatever

File: modules/015-admission-policy-engine/crds/doc-ru-security-policy.yaml [hunk 2]
```diff
@@ -192,6 +192,9 @@ spec:
                           publicKeys:
                             description: |
                               Список публичных ключей Cosign.
+                          registryCA:
```
Comment (yalosev) [inline modules/015-admission-policy-engine/crds/doc-ru-security-policy.yaml:195]:
```suggestion
                          ca:
```

**Review by yalosev — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Cherry pick PR [10946](https://github.com/deckhouse/deckhouse/pull/10946) to the branch [release-1.66](https://github.com/deckhouse/deckhouse/tree/release-1.66) successful!

**Pull request merged**